packages/core/src/interpreter.ts

```diff
@@ -503,6 +503,11 @@ export class Interpreter<
       this.doneListeners.delete(listener);
     }
 
+    if (!this.initialized) {
+      // Interpreter already stopped; do nothing
+      return this;
```

Comment:
Shouldn't we throw an error that just would lead users to fix their mistake? isn't stopping twice breaking an invariant rule of XState? I see arguments both ways (throwing vs making this noop) and I'm somewhat undecided what's better myself - just raising this as a general question.

Reply:
There can exist use-cases where the not-started interpreter is stopped by no fault of the developer; throwing would be counter-productive and unexpected.

Contrived example: in React, a component that is rendered but immediately unmounted (for some reason) would need to `.stop()` all services, even if they haven't started yet.

I think the best thing to do is to no-op, since the intention is to stop the service, and if the service is already stopped, mission accomplished 👍

Reply:
> Contrived example: in React, a component that is rendered but immediately unmounted (for some reason) would need to .stop() all services, even if they haven't started yet.

This actually can't happen - if we don't reach the commit phase we won't `start` anything **and** unmounting just wont happen so `stop` won't be called either.